Source/Scripts/SyntaxDeserializerGenerator.cs

```diff
@@ -61,7 +62,7 @@ protected override void HandleClass(Type type) {
       ownedFieldPosition = ParameterToSchemaPositions[baseType].Count;
     }
     var parameterToSchemaPosition = new Dictionary<string, int>();
-    var schemaToConstructorPosition = new Dictionary<int, int>();
+    var schemaToConstructorPosition = new SortedDictionary<int, int>();
```

Comment:
What's this for?

Reply:
Indices for `NonSerializedField` fields should not appear in this dictionary, so its keys may not be contiguous, and so to more easily iterate over its entries in order I changed this to a `SortedDictionary`. But I think it makes more sense to enforce the right iteration order at the point of iteration, rather than here at the point of construction, so I've reverted this type back to `Dictionary`.